### PR TITLE
more specific delete message for domain records

### LINKED ISSUE #1900: Be more explicit when deleting DNS records from a domain
Deleting a single record from a domain reads like the whole domian is about to be deleted.

delete-cname-record-modal

### PR DESCRIPTION
**Pull request by na3d**
Closes #1900

### CODE REVIEW

File: components/modals/DeleteModalBody.js [hunk 1]
```diff
@@ -19,6 +20,13 @@ export default function DeleteModalBody(props) {
         <p>This operation cannot be undone.</p>
       </div>
     );
+  } else if (parentItem) {
+    body = (
+      <p>
+        Are you sure you want
+        to <strong>permanently</strong> delete <strong>{items[0]}</strong> from {parentItem}?
```
Comment (mparke) [inline components/modals/DeleteModalBody.js]: if the above block supports multiple items, this too needs to support multiple items?

Comment (na3d) [inline components/modals/DeleteModalBody.js]: There is no situation yet where it needs to handle multiple items with showing a parent child relationship.

File: src/domains/components/MasterZone.js [hunk 2]
```diff
@@ -373,7 +373,8 @@ export class MasterZone extends Component {
                   cellComponent: ButtonCell,
                   headerClassName: 'ButtonColumn',
                   text: 'Delete',
-                  onClick: ({ id }) => this.renderDeleteRecord('Delete CNAME Record', id),
+                  onClick: ({ name, id }) =>
+                    this.renderDeleteRecord('Delete CNAME Record', id, name),
```
Comment (na3d) [inline src/domains/components/MasterZone.js]: The problem is, the CNAMEs are unique, nothing else is.

**Comment by eatonphil** [comment]
@na3d what I mean is this:

screen shot 2017-05-31 at 9 59 29 am

**Comment by na3d** [comment]
Ah, get rid of the parent child and just generally say record.  Got it.

File: components/modals/DeleteModalBody.js (hunk 1 above)
Comment (mparke) [inline components/modals/DeleteModalBody.js]: well, also look at the logic here, this will stop accounting for parent items as soon as we do have a case, which could occur in a completely unrelated change, because this is a generic component.

needs to be more flexible since the first block already checks for multiple items

File: src/domains/components/MasterZone.js [hunk 3]
```diff
@@ -32,13 +32,15 @@ export class MasterZone extends Component {
       target: ns,
       ttl_sec: ONE_DAY,
       name: domain.domain,
+      readOnly: true,
```
Comment (eatonphil) [inline src/domains/components/MasterZone.js]: Is this for all NS records? Not all NS records are readonly... Only the ones that follow the logic above that you remove are read only.

Comment (na3d) [inline src/domains/components/MasterZone.js]: Only the default NS1-5 are read only.  If someone makes a duplicate of one of those, it isn't read only, which was getting caught with the old way of handling this.

**Comment by eatonphil** [comment]
tested. looks good.

File: components/modals/DeleteModalBody.js [hunk 4]
```diff
@@ -8,6 +8,7 @@ export default function DeleteModalBody(props) {
     items,
     onCancel,
     typeOfItem,
+    parentItem,
```
Comment (eatonphil) [inline components/modals/DeleteModalBody.js]: This shouldn't be here.
